### PR TITLE
[berkeley] Expose lookup commitments

### PR DESCRIPTION
**Pull request by dannywillems**
Waiting for https://github.com/MinaProtocol/mina/pull/13899 to be merged to continue.
FIx https://github.com/MinaProtocol/mina/issues/13909
Depends on https://github.com/o1-labs/snarkyjs/pull/1089

- [x] ~~Regenerate bindings when 13899 is merged.~~. Artefacts are built on top of it.

### CODE REVIEW

File: kimchi/js/bindings.js [hunk 1]
```diff
@@ -2353,11 +2380,37 @@ var caml_pasta_fp_commitments_to_rust = function (x) {
...
+
+  var sorted = convertArray(wasm_lc.sorted);
+  var aggreg = caml_vesta_poly_comm_of_rust(wasm_lc.aggreg);
+  var wasm_lc_runtime = wasm_lc.runtime;
+  var caml_runtime;
+  if (wasm_lc_runtime === undefined) {
+    caml_runtime = 0;
+  } else {
+    caml_runtime = [0, caml_vesta_poly_comm_of_rust(wasm_lc_runtime)];
+  }
```
Comment (rbonichon) [inline kimchi/js/bindings.js:2407]: I think there might be a `opt_of_rust` function that does this as a one-liner.

Comment (rbonichon) [reply kimchi/js/bindings.js:2407]: You're actually using it below.

File: kimchi/js/bindings.js [hunk 2]
```diff
@@ -2591,11 +2673,37 @@ var caml_pasta_fq_commitments_to_rust = function (x) {
...
+  };
+
+  var sorted = convertArray(wasm_lc.sorted);
+  var aggreg = caml_pallas_poly_comm_of_rust(wasm_lc.aggreg);
+  var wasm_lc_runtime = wasm_lc.runtime;
+  var caml_runtime;
+  if (wasm_lc_runtime === undefined) {
+    caml_runtime = 0;
+  } else {
+    caml_runtime = [0, caml_pallas_poly_comm_of_rust(wasm_lc_runtime)];
```
Comment (rbonichon) [inline kimchi/js/bindings.js:2699]: `caml_opt_of_rust` possibly useful

File: kimchi/wasm/src/plonk_proof.rs [hunk 3]
```diff
@@ -223,8 +332,7 @@ macro_rules! impl_proof {
                         w_comm: array_init(|i| x.w_comm[i].clone().into()),
                         z_comm: x.z_comm.clone().into(),
                         t_comm: x.t_comm.clone().into(),
-                        // TODO
-                        lookup: None,
+                        lookup: x.lookup.clone().map(Into::into),
```
Comment (rbonichon) [inline kimchi/wasm/src/plonk_proof.rs:335]: Nice to see this todo fixed. Thanks!

**Review by rbonichon — APPROVED** [review]
LGTM.  I only have very minor improvements

File: kimchi/js/bindings.js (hunk 1 above)
Comment (dannywillems) [reply kimchi/js/bindings.js:2407]: Yes, it seems that even in develop I haven't done it. I'm opening a follow-up issue to make both branch the same.

File: kimchi/js/bindings.js (hunk 2 above)
Comment (dannywillems) [reply kimchi/js/bindings.js:2699]: Same than above.

**Comment by dannywillems** [comment]
Opened follow-up issue [here](https://github.com/MinaProtocol/mina/issues/13944) regarding @rbonichon comments above.

**Pull request merged**